lib/tramway/base_form.rb

```diff
@@ -20,38 +25,10 @@ def initialize(object)
     class << self
       def inherited(subclass)
         subclass.instance_variable_set(:@properties, [])
+        subclass.instance_variable_set(:@normalizations, {})
```

Comment:
are you sure that it's possible? `inherited` method can be defined once as I know.

Reply:
Just made methods like `__initialize_properties`, etc. is it ok?

Reply:
It is better, yes, and could be a good stop here.
It would be better to move `__initialize_properties` inside mixin too. There could be several solutions, it seems `prepend` could be useful here (but I haven't used it)

```
module Tramway
  module Forms
    module Properties
      def self.included(base)
        base.extend ClassMethods
        base.prepend InheritanceMethods
      end

      module InheritanceMethods
        def inherited(subclass)
          __initialize_properties(subclass)
          super
        end
      end

      module ClassMethods
        def __initialize_properties(subclass)
          # Implementation of __initialize_properties
        end
      end
    end
  end
end
```

Reply:
Yes, we stop here. But let's discuss this example.

I didn't check it, but looks like it won't work cuz `inherited` from `Properties` will be overridden by `inherited` from `Normalizations`

Reply:
It should not override because it uses `base.prepend`, and it'll add `inherited` to the method lookup chain. Simplified example here https://replit.com/@Khelek/DismalHightechJavascript